### PR TITLE
Update agent setup instructions

### PR DESCRIPTION
**Pull request by ravikiran-sulikeri**
----

If you open a PR that needs to go into a current version, you need to *cherry-pick your commit from dev over to the current version branch*. Only then will the proper builds that generate html/pdf be run. But beware: Docs will be generated but not published automatically!

- [ ] N/A - or - I have added the appropriate "cherry-pick-to" labels to this PR so I don't forget to do this later!

### CODE REVIEW

File: modules/ROOT/pages/addition/agent-installation/index.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/index.adoc]: I worry it might be a bit confusing that some of the info is on the root-level page (Installation and configuration of NOM agents) and some is on a sub-page (the registration stuff).

Maybe it would be clearer if the root-level page was just the content above this comment, essentially a a brief description and then a bullet list of steps, like:
1. Register agent
  There are two ways:
  a) Manual registration - recommended for manually..
  b) Self-registered - recommended for automated..
2. Configure and run agent

And then the bulk of the content on index.adoc is moved into the second bullet. Thoughts?

Comment (SeBBBe) [reply modules/ROOT/pages/addition/agent-installation/index.adoc]: OR maybe there should be 3 top-level sections like:

* Neo4j instance requirements
* Agent registration
* Agent configuration

File: modules/ROOT/pages/addition/agent-installation/install-packaged.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/install-packaged.adoc]: Missing the actual steps?

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc:135]: Is there a reason we're not recommending self-signed certs for mTLS in production? I get it with HTTPS certs that you'd want to have a proper CA signed cert, but for self-registration it shouldn't matter as long as the key is kept safe. Unless I'm missing something?

File: modules/ROOT/pages/addition/agent-installation/index.adoc
Comment (ravikiran-sulikeri) [reply modules/ROOT/pages/addition/agent-installation/index.adoc]: Yeah I am restructuring this section in a similar pattern, just weighing clarity against brevity, if duplicating same steps woun't hurt

File: modules/ROOT/pages/addition/agent-installation/install-packaged.adoc
Comment (ravikiran-sulikeri) [reply modules/ROOT/pages/addition/agent-installation/install-packaged.adoc]: Still working out best way to unpack and run agents within a container, will update

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (ravikiran-sulikeri) [reply modules/ROOT/pages/addition/agent-installation/self-registered.adoc:135]: You are right about the agent keys being a maintenance overhead, we can still suggest it though

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]:
```suggestion
Agent self-registration omits manually acquiring initial configuration from NOM UI as described in xref:./manual.adoc#register[Manually registering agent].
Instead minimal generic configuration about the NOM server is provided and the additional agent specific configuration is obtained from the NOM server after authorization. 
Authorization is obtained either through mutual authentication or manual authorization of the agent from UI. 
```

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]:
```suggestion
Once the agent is approved, the agent will receive initial configuration from NOM server which it persists on the instance host in a file named `nom-agent-config.yaml`.
```

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]: Does this have to be done before approval to be correctly persisted or will it still be persisted if it's done later?

If it can be done later - I think we should move this step to the end to keep the flow simple.

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]: What if no instances are configured? 
What if multiple instances are configured?

Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]:
```suggestion
| `NEO4J_CONF` if set or `conf` folder under `NEO4J_HOME` if set, else `.nom` folder in user home directory.
```

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc:89]: Can this additional meta-data be provided in the mtls set up too?

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]:
```suggestion
Agent meta-data can be optionally specified using these configuration parameters:
```

File: modules/ROOT/pages/addition/agent-installation/index.adoc
Comment (eve-bright) [reply modules/ROOT/pages/addition/agent-installation/index.adoc]: I agree that restructuring would be good. I don't think we should worry too much about repetition if it improves clarity. 
In particular, I think it needs to be clearer that the instance configuration will need to be specified regardless of the registration strategy. 
Similarly I think it needs to be clearer how to run the agent in different modes (service, console) with the different strategies.

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (ravikiran-sulikeri) [reply modules/ROOT/pages/addition/agent-installation/self-registered.adoc]: it can be done anytime

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (ravikiran-sulikeri) [reply modules/ROOT/pages/addition/agent-installation/self-registered.adoc]: No instances would result in config being saved to user home directory. for multiple instances, since agent looks for current env for config folder path, it could only be one value or no value per host right?

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (ravikiran-sulikeri) [reply modules/ROOT/pages/addition/agent-installation/self-registered.adoc:89]: yes

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (SeBBBe) [reply modules/ROOT/pages/addition/agent-installation/self-registered.adoc]: I think we should remove this step from the instruction in the interest of brevity - would think that the ability to change the name would be obvious enough.

File: modules/ROOT/pages/installation/server.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/installation/server.adoc:279]: I just realized I used the incorrect casing for the analytics stuff here as well, but we don't use camel case in any other the other parameters. Better would be `trust-cert-collection` (yes, I'm aware it's a BE change)

File: modules/ROOT/pages/installation/docker/container.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/installation/docker/container.adoc]:
```suggestion
== Getting the NOM server Docker image
The latest NOM server image can be pulled from the verified repository by running the following command.
```

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]:
```suggestion
. Upon approval, the agent status changes to `Offline` until the agent connects to the server with the initial configuration.
```

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]:
```suggestion
NOTE: An unauthorized self-registering agent will have a default name derived from its host information. It's recommended to update this name or use the optional configuration described <<agent-meta-data,here>>.
```

**Review by SeBBBe — COMMENTED** [review]
The new layout is much easier to follow! Some minor comments.

File: modules/ROOT/pages/addition/agent-installation/agent-config-file.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/agent-config-file.adoc]:
```suggestion
The agent configuration file is used by a self-registering agent to persist its initial configuration, which is required for restarts.
```

File: modules/ROOT/pages/addition/agent-installation/agent-config-file.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/agent-config-file.adoc]:
```suggestion
Agent initially reports an error while running in self-register mode such as `| ERROR | unable to load config from file error="open /home/.nom/nom-agent-config.yaml: no such file or directory"`. This is expected behavior, as the file will be created automatically.
```

Comment (SeBBBe) [reply modules/ROOT/pages/addition/agent-installation/agent-config-file.adoc]: I think we can simplify this a bit

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]: This sort of makes it sound like the initial configuration will not be persisted unless the optional parameter is used 🤔

Comment (SeBBBe) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]:
```suggestion
The following optional configuration can be used to specify the location for agent to save its initial configuration:
```

File: modules/ROOT/pages/addition/kubernetes/daemonset.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/kubernetes/daemonset.adoc:9]: Is there anything else which needs to be setup or is disk usage the only metric we can't grab by default? 🤔

File: modules/ROOT/pages/addition/kubernetes/index.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/kubernetes/index.adoc]:
```suggestion
Having a transparent view of the host where the Neo4j instance is deployed is a functional requirement for NOM agent. This view includes the ability of NOM agent to monitor and collect information about host resources such as disk, memory and cpu. In environments like a Kubernetes cluster, process containment makes it difficult to get a view of actual resource usage. 
```

File: modules/ROOT/pages/addition/kubernetes/index.adoc
Comment (SeBBBe) [inline modules/ROOT/pages/addition/kubernetes/index.adoc]:
```suggestion
Neo4j deployments in Kubernetes environment are supported through Helm charts. To monitor a Neo4j instance as a container in a Kubernetes environment, there are several options with varying degrees of support for NOM agent.
```

File: modules/ROOT/pages/addition/kubernetes/daemonset.adoc
Comment (ravikiran-sulikeri) [reply modules/ROOT/pages/addition/kubernetes/daemonset.adoc:9]: A Daemonset should be able to report Node's memory, cpu usage. Other than that only the disk usage that would be different for Neo4j pod since it can be mounted as an external file system.

Comment (SeBBBe) [reply modules/ROOT/pages/addition/kubernetes/daemonset.adoc:9]: Great!

**Review by SeBBBe — APPROVED** [review]

File: modules/ROOT/pages/addition/agent-installation/self-registered.adoc
Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/self-registered.adoc]:
```suggestion
WARNING: agent config location must be of persistent type. 
```

File: modules/ROOT/pages/addition/agent-installation/manual.adoc
Comment (eve-bright) [inline modules/ROOT/pages/addition/agent-installation/manual.adoc]:
```suggestion
= Manually registered agent
```

**Review by eve-bright — APPROVED** [review]

**Pull request merged**
